### DISCUSSION TITLE
Help with a recolor of a folder of images to a set of hex colors.

### DISCUSSION THREAD

**Discussion by leegweaver** [ImageMagick/ImageMagick#4625] [Digital Image Processing] [answered]
I'm using the Windows version,  and I was pointed here for some help,  the person that sent me here knew how to do what I need to do but only using Linux.

what we need to know is how to batch process a directory of images and recolor them all outputting 118 different images from a set of 118 Hex color codes for each original image in the directory.   Hopefully naming each originalfileanmehexcode.png when done  if there is a tutorial for something like this out there I  would be glad to read and or watch it.  Lacking that I would be appreciative of any help anyone here would be willing to give.

**Comment by leegweaver** [comment]
I have come up with come code that is close I think to what I need,  but I need to strip out the original file extension from my %%a variable. 
and I have tried several ways that I have seen but I can't get it working.  any help would be greatly appreciated.

```
echo off
SETLOCAL EnableDelayedExpansion
for /f "tokens=*" %%x in (118_hex_colors.txt) do (
	for /r %%a in (*.png) do ( 
	echo %%a%%x.png
        Rem above line to ensure I'm passing the correct final filename to image magick
		rem magick %%a -colorspace gray -fill "#%%x" -tint 100 %%a%%x.png

	)
 )
```

**Comment by snibgo** [comment, marked as answer]
> ... but I need to strip out the original file extension from my %%a variable.

Type "help for" at the command line. At the end, that documentation gives the answer: use "%%~na".

**Reply by leegweaver** [reply]
thank you I must be blind.
